stan/math/prim/functor/hcubature.hpp

```diff
@@ -163,17 +162,21 @@ inline void increment(std::vector<bool>& index, const int k,
  * @param p vector of vectors
  */
 inline void signcombos(const int k, const double lambda, const int dim,
-                       std::vector<std::vector<double>>& p) {
+                       Eigen::MatrixXd& p) {
   std::vector<int> c(k);
   const auto choose_dimk = choose(dim, k);
+  p.resize(dim, choose_dimk * std::pow(2, k));
+  p.setZero();
+  int col = 0;
   for (std::size_t i = 1; i != choose_dimk + 1; i++) {
-    std::vector<double> temp(dim, 0.0);
+    Eigen::VectorXd temp = Eigen::VectorXd::Zero(dim);
     combination(c, dim, k, i);
     std::vector<bool> index;
     index.clear();
     for (std::size_t j = 0; j != std::pow(2, k); j++) {
       increment(index, k, lambda, c, temp);
-      p.push_back(temp);
+      p.col(col) = temp;
+      col += 1;
```

Comment:
Oh actually is there a reason you cannot use `p` directly instead of `temp`? Also I'd rather not have `.col(col)`, just sort of weird naming can we call the int `col` something else?